pkg/handler/v1/provider/openstack.go

```diff
@@ -28,34 +29,45 @@ import (
 	"k8c.io/kubermatic/v2/pkg/provider"
 )
 
+type credential struct {
+	credential, username, password, tenant, tenantID, domain string
+}
+
+func auth(ctx context.Context, request OpenstackReq, userInfoGetter provider.UserInfoGetter, presetsProvider provider.PresetProvider) (*provider.UserInfo, *credential, error) {
+	req := reflect.ValueOf(request)
```

Comment:
You don't need to use `reflect` any longer, right?

Reply:
right